api/services/Reddit.js

```diff
@@ -47,20 +49,18 @@ var makeRequest = async function (refreshToken, requestType, url, data, rateLimi
     method: requestType,
     formData: data
   };
-  let response = await request(options);
+  let response = await request(options).catch(function (error) {
+    console.log('Reddit error: ' + requestType + ' request sent to ' + url + ' returned ' + error.statusCode +
+      ' - ' + error.statusMessage + '\nForm data sent: ' + JSON.stringify(data));
+    throw {statusCode: 502, error: 'Reddit responded with status code ' + error.statusCode};
+  });
```

Comment:
The `request-promise` package already rejects if the response status code is something other than 2xx, so our other error-handling code wasn't getting used. I moved it up here instead.